// mat-table-exporter/esm5/lib/mat-table-exporter.module.js
import { __decorate } from "tslib";
import { CdkTableExporterModule } from 'cdk-table-exporter';
import { MatTableModule } from '@angular/material/table';
import { NgModule } from '@angular/core';
import { MatTableExporterDirective } from './mat-table-exporter.directive';
import * as ɵngcc0 from '@angular/core';
var MatTableExporterModule = /** @class */ (function () {
    function MatTableExporterModule() {
    }
MatTableExporterModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: MatTableExporterModule });
MatTableExporterModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function MatTableExporterModule_Factory(t) { return new (t || MatTableExporterModule)(); }, imports: [[
            MatTableModule,
            CdkTableExporterModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(MatTableExporterModule, { declarations: function () { return [MatTableExporterDirective]; }, imports: function () { return [MatTableModule,
        CdkTableExporterModule]; }, exports: function () { return [MatTableExporterDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(MatTableExporterModule, [{
        type: NgModule,
        args: [{
                declarations: [MatTableExporterDirective],
                imports: [
                    MatTableModule,
                    CdkTableExporterModule
                ],
                exports: [MatTableExporterDirective]
            }]
    }], function () { return []; }, null); })();
    return MatTableExporterModule;
}());
export { MatTableExporterModule };
