// ng-marquee/esm5/lib/ng-marquee.module.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/ng-marquee.module.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule } from '@angular/core';
import { NgMarqueeComponent } from './ng-marquee/ng-marquee.component';
import { CommonModule } from '@angular/common';
import * as ɵngcc0 from '@angular/core';
var NgMarqueeModule = /** @class */ (function () {
    function NgMarqueeModule() {
    }
NgMarqueeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgMarqueeModule });
NgMarqueeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgMarqueeModule_Factory(t) { return new (t || NgMarqueeModule)(); }, imports: [[
            CommonModule,
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgMarqueeModule, { declarations: function () { return [NgMarqueeComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [NgMarqueeComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgMarqueeModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule,
                ],
                declarations: [NgMarqueeComponent],
                exports: [NgMarqueeComponent]
            }]
    }], function () { return []; }, null); })();
    return NgMarqueeModule;
}());
export { NgMarqueeModule };
