// cdk-table-exporter/esm2015/lib/cdk-table-exporter.module.js
import { __decorate } from "tslib";
import { CdkTableModule } from '@angular/cdk/table';
import { NgModule } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
let CdkTableExporterModule = class CdkTableExporterModule {
};
CdkTableExporterModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: CdkTableExporterModule });
CdkTableExporterModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function CdkTableExporterModule_Factory(t) { return new (t || CdkTableExporterModule)(); }, imports: [[
            CdkTableModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(CdkTableExporterModule, { imports: function () { return [CdkTableModule]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CdkTableExporterModule, [{
        type: NgModule,
        args: [{
                declarations: [],
                imports: [
                    CdkTableModule
                ],
                exports: []
            }]
    }], null, null); })();
export { CdkTableExporterModule };
